libs/libc/math/lib_round.c

```diff
@@ -34,6 +34,11 @@
 #ifdef CONFIG_HAVE_DOUBLE
 double round(double x)
 {
+  if (isinf(x) || isnan(x))
+    {
+      return (x);
+    }
+
   double f = modf(x, &x);
```

Comment:
Please separate this. Move `double f;` before `if (isinf(x) || isnan(x))` abd change here to `f = modf(x, &x);`

Reply:
please address this. It is C89 incompliant